src/main/drivers/accgyro/accgyro_spi_icm40609.c

```diff
@@ -0,0 +1,769 @@
+
+        if (fabsf(coswz) <= 0.875f) {
+            int16_t nf_coswz = (int16_t)roundf(coswz * 256.0f);
+            nf_coswz_lsb = nf_coswz & 0xFF;
+            nf_coswz_msb = (nf_coswz >> 8) & 0x01;
+            nf_coswz_sel = 0;
+        } else {
+            nf_coswz_sel = 1;
+            int16_t nf_coswz;
+            if (coswz > 0.875f) {
+                nf_coswz = (int16_t)roundf(8.0f * (1.0f - coswz) * 256.0f);
+            } else {
+                nf_coswz = (int16_t)roundf(8.0f * (1.0f + coswz) * 256.0f);
+            }
+            nf_coswz_lsb = nf_coswz & 0xFF;
+            nf_coswz_msb = (nf_coswz >> 8) & 0x01;
+        }
+
+        // Write NF_COSWZ[7:0] for X, Y, Z
+        spiWriteReg(dev, ICM40609_REG_GYRO_CONFIG_STATIC6, nf_coswz_lsb); // X
+        spiWriteReg(dev, ICM40609_REG_GYRO_CONFIG_STATIC7, nf_coswz_lsb); // Y
+        spiWriteReg(dev, ICM40609_REG_GYRO_CONFIG_STATIC8, nf_coswz_lsb); // Z
+
+        // Write NF_COSWZ[8] and NF_COSWZ_SEL into STATIC9 (0x12)
+        uint8_t reg9 = 0;
+        reg9 |= (nf_coswz_msb << 0); // X NF_COSWZ[8]
+        reg9 |= (nf_coswz_msb << 1); // Y NF_COSWZ[8]
+        reg9 |= (nf_coswz_msb << 2); // Z NF_COSWZ[8]
+        reg9 |= (nf_coswz_sel << 3); // X COSWZ_SEL
+        reg9 |= (nf_coswz_sel << 4); // Y COSWZ_SEL
+        reg9 |= (nf_coswz_sel << 5); // Z COSWZ_SEL
+        spiWriteReg(dev, ICM40609_REG_GYRO_CONFIG_STATIC9, reg9);
+    }
+
+    icm40609SelectUserBank(dev, ICM40609_USER_BANK_0);
+}
+
+static void icm40609SetTempFiltBw(const extDevice_t *dev, icm40609TempFiltBw_e bw)
+{
+    icm40609SelectUserBank(dev, ICM40609_USER_BANK_0);
+
+    uint8_t reg51 = spiReadRegMsk(dev, ICM40609_REG_GYRO_CONFIG1);
+    reg51 &= ~(0x07 << 5);
+    reg51 |= (bw << 5);
+    spiWriteReg(dev, ICM40609_REG_GYRO_CONFIG1, reg51);
+}
+
+static void icm40609SetGyroUiFiltOrder(const extDevice_t *dev, icm40609UiFiltOrder_e order)
+{
+    icm40609SelectUserBank(dev, ICM40609_USER_BANK_0);
+
+    uint8_t reg51 = spiReadRegMsk(dev, ICM40609_REG_GYRO_CONFIG1);
+    reg51 &= ~ICM40609_GYRO_UI_FILT_ORD_MASK;
+    reg51 |= (order << 2); // Write GYRO_UI_FILT_ORD to bits [3:2]
+    spiWriteReg(dev, ICM40609_REG_GYRO_CONFIG1, reg51);
+}
+
+static void icm40609SetAccelUiFiltOrder(const extDevice_t *dev, icm40609UiFiltOrder_e order)
+{
+    icm40609SelectUserBank(dev, ICM40609_USER_BANK_0);
+
+    uint8_t reg53 = spiReadRegMsk(dev, ICM40609_REG_GYRO_ACCEL_CONFIG1);
+    reg53 &= ~ICM40609_ACCEL_UI_FILT_ORD_MASK;
+    reg53 |= (order << 3); // Write ACCEL_UI_FILT_ORD to bits [4:3]
+    spiWriteReg(dev, ICM40609_REG_GYRO_ACCEL_CONFIG1, reg53);
+}
+
+static void icm40609SetGyroDec2M2(const extDevice_t *dev, bool enable)
+{
+    icm40609SelectUserBank(dev, ICM40609_USER_BANK_0);
+
+    uint8_t reg51 = spiReadRegMsk(dev, ICM40609_REG_GYRO_CONFIG1);
+    reg51 &= ~ICM40609_GYRO_DEC2_M2_ORD_MASK;
+
+    if (enable) {
+        reg51 |= (2 << 0);
+    } else {
+        reg51 |= (0 << 0);
```

Comment:
00 seems to be reserved:

Selects order of GYRO DEC2_M2 Filter
00: Reserved
01: Reserved
10: 3rd Order
11: Reserved